code/datums/gamemode/factions/bloodcult/bloodcult_buildings.dm

```diff
@@ -1407,8 +1407,9 @@ var/list/bloodstone_list = list()
 	anchor = FALSE
 	for (var/obj/structure/teleportwarp/TW in src.loc)
 		qdel(TW)
-	new /obj/machinery/singularity/narsie/large(src.loc)
-	stat_collection.cult_narsie_summoned = TRUE
+	if (src && loc)
```

Comment:
Src can never be null, you'd be better off checking if(!gcDestroyed && loc)

This is evidence of a worse problem involving the while loop never properly breaking. Go to 1382 and change it to while(!gcDestroyed && loc && anchor)

Honestly, this is rather gunked and may need some rework